presto-hive/src/main/java/com/facebook/presto/hive/ManifestPartitionLoader.java

```diff
@@ -0,0 +1,149 @@
+import static com.facebook.presto.hive.HiveManifestUtils.getFileNames;
+import static com.facebook.presto.hive.HiveSessionProperties.getMaxInitialSplitSize;
+import static com.facebook.presto.hive.HiveSessionProperties.getMaxSplitSize;
+import static com.facebook.presto.hive.HiveSessionProperties.getNodeSelectionStrategy;
+import static com.facebook.presto.hive.HiveUtil.getInputFormat;
+import static com.facebook.presto.hive.metastore.MetastoreUtil.getPartitionLocation;
+import static com.google.common.base.Verify.verify;
+import static com.google.common.collect.ImmutableList.toImmutableList;
+import static java.util.Objects.requireNonNull;
+
+public class ManifestPartitionLoader
+        extends PartitionLoader
+{
+    private final Table table;
+    private final Optional<Domain> pathDomain;
+    private final ConnectorSession session;
+    private final HdfsEnvironment hdfsEnvironment;
+    private final HdfsContext hdfsContext;
+    private final boolean schedulerUsesHostAddresses;
+
+    public ManifestPartitionLoader(
+            Table table,
+            Optional<Domain> pathDomain,
+            ConnectorSession session,
+            HdfsEnvironment hdfsEnvironment,
+            boolean schedulerUsesHostAddresses)
+    {
+        this.table = requireNonNull(table, "table is null");
+        this.pathDomain = requireNonNull(pathDomain, "pathDomain is null");
+        this.session = requireNonNull(session, "session is null");
+        this.hdfsEnvironment = requireNonNull(hdfsEnvironment, "hdfsEnvironment is null");
+        this.hdfsContext = new HdfsContext(session, table.getDatabaseName(), table.getTableName());
+        this.schedulerUsesHostAddresses = schedulerUsesHostAddresses;
+    }
+
+    public ListenableFuture<?> loadPartition(HivePartitionMetadata partition, HiveSplitSource hiveSplitSource, boolean stopped)
+            throws IOException
+    {
+        Path path = new Path(getPartitionLocation(table, partition.getPartition()));
+        Map<String, String> parameters = partition.getPartition().get().getParameters();
+        List<String> fileNames = getFileNames(parameters.get(FILE_NAMES));
+        List<Long> fileSizes = Arrays.stream(parameters.get(FILE_SIZES).split(",")).map(Long::valueOf).collect(toImmutableList());
+
+        // Verify that the count of fileNames and fileSizes are same
+        verify(fileNames.size() == fileSizes.size(), "List of fileNames and fileSizes differ in length");
+
+        ImmutableList.Builder<HiveFileInfo> fileListBuilder = ImmutableList.builder();
+        for (int i = 0; i < fileNames.size(); i++) {
+            Path filePath = new Path(path, fileNames.get(i));
+            FileStatus fileStatus = new FileStatus(fileSizes.get(i), false, 1, getMaxSplitSize(session).toBytes(), 0, filePath);
+            try {
+                BlockLocation[] locations = new BlockLocation[] {new BlockLocation(new String[] {"localhost:50010"}, new String[] {"localhost"}, 0, fileSizes.get(i))};
```

Comment:
Do we really need this? This could be an empty array right?

Reply:
If its empty, its failing because of the check in InternalHiveSplitFactory.java
https://github.com/prestodb/presto/blob/6970b6da9f392a8f6eb41c974c7eeb15f69c62da/presto-hive/src/main/java/com/facebook/presto/hive/util/InternalHiveSplitFactory.java#L181